ocaml/src/blob_access_factory.ml

```diff
@@ -0,0 +1,59 @@
+(*
+Copyright (C) 2016 iNuron NV
+
+This file is part of Open vStorage Open Source Edition (OSE), as available from
+
+
+    http://www.openvstorage.org and
+    http://www.openvstorage.com.
+
+This file is free software; you can redistribute it and/or modify it
+under the terms of the GNU Affero General Public License v3 (GNU AGPLv3)
+as published by the Free Software Foundation, in version 3 as it comes
+in the <LICENSE.txt> file of the Open vStorage OSE distribution.
+
+Open vStorage is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY of any kind.
+*)
+
+open Blob_access
+
+let make_directory_access files_path =
+  let r = new default_directory_access files_path in
+  (r :> directory_access)
+
+let _service_handle = ref None
+
+let maybe_init_service gioexecfile =
+  match !_service_handle with
+  | None ->
+    begin
+      let service_handle = Gobjfs.IOExecFile.init gioexecfile in
+      let align_size = 4096 in
```

Comment:
I'll introduce a constant iso all the floating 4096's. But yes, it's weird we pull this out of thin air